sentence_transformers/SentenceTransformer.py

```diff
@@ -797,6 +768,9 @@ def _save_checkpoint(self, checkpoint_path, checkpoint_save_total_limit, step):
                 old_checkpoints = sorted(old_checkpoints, key=lambda x: x['step'])
                 shutil.rmtree(old_checkpoints[0]['path'])
 
+        if save_to_hub:
+            self.save_to_hub(**self.hub_parameters, exist_ok=True, commit_message="Push new training checkpoint.")
```

Comment:
What would go in `hub_parameters` and what would go in normal params? E.g. why is `exist_ok` separate?

Reply:
Ah I should have documented it better. `hub_parameters` will be passed entirely to `save_to_hub`, so users should only put in parameters that make sense for `save_to_hub`. However, I have to override `exist_ok` so the call to `create_repo` in `save_to_hub` won't fail after the first checkpoint.

Otherwise I'm not quite sure what pattern to use to do it cleanly without making structural changes to the code. I could create a `HubParameters` class to hold the parameters, or I could have users pass all of the hub parameters when `.fit()` is called? `fit` already has a ton of parameters, so I didn't want to add more in there.

Reply:
Makes sense to have a `HubParameters`